erpnext/templates/includes/cart.js

```diff
@@ -24,6 +24,16 @@ $.extend(shopping_cart, {
 			if($(this).prop("checked")) {
 				var me = this;
 
+				// freeze ui and wait for response:
+				frappe.freeze();
```

Comment:
@robulik,

instead of using frappe.freeze() explicitly you can pass <code>freeze: true</code> in frappe.call it will handle the freeze and unfreeze event